pkg/controllers/cluster/controller.go

```diff
@@ -202,3 +213,32 @@ func (h *handler) createNamespace(cluster *fleet.Cluster, status fleet.ClusterSt
 	}
 	return err
 }
+
+func (h *handler) reconcileAgentEnvVars(cluster *fleet.Cluster) error {
+	hashed, err := hashAgentEnvVars(cluster.Spec.AgentEnvVars)
+	if err != nil {
+		return err
+	}
+	if cluster.Labels[agentEnvVarsHashLabel] != hashed {
+		// We enqueue to ensure that we edit the labels after other controllers.
+		h.namespaces.Enqueue(cluster.Namespace)
+		if cluster.Labels == nil {
+			cluster.Labels = make(map[string]string, 1)
+		}
+		cluster.Labels[agentEnvVarsHashLabel] = hashed
+	}
+	return nil
+}
+
+func hashAgentEnvVars(agentEnvVar []v1.EnvVar) (string, error) {
+	if len(agentEnvVar) < 0 {
```

Comment:
We need to check len(agentEnvVar) == 0. And If there is no agentEnv, hash should be no-op and we shouldn't be writing labels to clusters. This will give minimal impacts on majority of the clusters that don't have agentEnv set, cause json.marshall and hash function are not cheap.

Reply:
Few accidental mistakes there. Good catch.